### ISSUE TITLE
Make some dependencies optional

### ISSUE THREAD

**Issue by FHU-yezi** [pynecone-io/pynecone#73]
Just installed this library use Poetry and `poetry show --tree` seems to be that:

```
pynecone 0.0.70 Build your CLI with ease
├── certifi *
├── jinja2 *
│   └── markupsafe >=2.0
├── keyring *
│   ├── (some deps)
├── pika *
├── python-dotenv *
├── pyyaml *
├── requests *
│   ├── (some-deps)
├── requests-toolbelt *
│   └── (some deps)
├── sqlalchemy *
│   └── greenlet !=0.4.17
└── tabulate *
```

When I want to use MongoDB as my database, I don't use the internal ORM provided by Pynecone, and the dependence `SQLAlchemy` is not nesserary for me. We can detect if user installed this library when they try to import a module which need it, and this dependence should not be included by default.

`extras` in `pyproject.toml` may be helpful for this, we can install the ORM-ready version with `pynecone[orm]`, and all the optional dependencies installed version with `pynecone[all]`.

By the way, `httpx` is a better replacement for `requests`, it support HTTP/2 and asyncio, and may be helpful for the support of async function in next versions. If you think that is a good idea, I will open another issue about it.

**Comment by picklelo** [comment]
This is a good point. I'll think about it - there's a tradeoff to be made between installing too much, but also I would like pynecone to work out of the box without any additional installs. It may lead to increased confusion and room for error.

(Also - I think you may have installed the wrong package? There is unfortunately another package called `pynecone`, ours is called `pynecone-io`)

I didn't know about `httpx`, I'm open to switching if it's better, go ahead and make a ticket for that one.

**Comment by FHU-yezi** [comment]
Oh, I switched to the right package :), and the `plotly` should also be optional.

I found `redis` in our package, what is use for? I just know is it a key-value database for caching etc, it's for the server-side state stone?

**Comment by Alek99** [comment]
> Oh, I switched to the right package :), and the `plotly` should also be optional.
> 
> I found `redis` in our package dependencies, what is use for? I just know is it a key-value database for caching etc, it's for the server-side state storage?

Yeah this is used in our pc prod for state session expiry, so that user sessions don't build up over time. We take this out though. your right running redis with pynecone should be optional and not in the core dependencies

**Comment by picklelo** [comment]
Yeah we use it for state management when running in prod mode, so the worker processes can share state.

We will think about separating it, I know it's not ideal to have extra installs but I also don't necessarily like the overhead of installing extra packages to access a specific feature of Pynecone.
